### PR TITLE
Fetch and display hard-coded comments data

### PR DESCRIPTION
**Pull request by fatimazali**
image

- Lay down basic structure for future comments feature
- Return hard-coded data from server in DataServlet.java
- Fetch data using JavaScript and display it on the page using HTML elements

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,46 @@
...
+    fatimaLocations.add("Irvine, California");
+
+    // Convert the locations to JSON
+    String json = convertToJsonUsingGson(fatimaLocations);
+   
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+    // response.setContentType("text/html;");
+    // response.getWriter().println("<h3>Hello Fatima!</h3>");
```
Comment (jessimb) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: nit: remove comments from production code

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,6 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/**
+ * Gets hard-coded locations from the server and
+  adds them to the DOM
+ */
```
Comment (jessimb) [inline portfolio/src/main/webapp/script.js:18]:
```suggestion
 * Gets hard-coded locations from the server and adds them to the DOM
 */
```

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,6 +12,36 @@
...
+async function getLocationsUsingAsyncAwait() { 
+
```
Comment (jessimb) [inline portfolio/src/main/webapp/script.js:20]: nit: remove extra newline, here and end of this function

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -31,3 +61,13 @@ function addRandomFact() {
...
 }
+
+
+/**
+ * Gets a hard-coded "Hello Fatima" greeting from the server and adds it to the DOM
+ */
+async function getGreetingUsingAsyncAwait() { 
+  const response = await fetch('/data');
+  const greeting = await response.text();
+  document.getElementById('greeting-container').innerHTML = greeting;
```
Comment (jessimb) [inline portfolio/src/main/webapp/script.js:72]: can innerHTML be innerText? innerHTML is vulnerable to XSS attacks

Comment (fatimazali) [reply portfolio/src/main/webapp/script.js:72]: (See Pull Request #7)

**Pull request merged**